projects/igniteui-angular/src/lib/tabs/tabs.component.ts

```diff
@@ -213,21 +229,32 @@ export class IgxTabsComponent implements IgxTabsBase, AfterViewInit, OnDestroy {
      */
     @HostListener('onTabItemSelected', ['$event'])
     public selectedGroupHandler(args) {
-        const prevSelectedIndex = this.selectedIndex;
-        if (prevSelectedIndex !== -1 && this.groups.toArray()[prevSelectedIndex] !== undefined) {
-            this.onTabItemDeselected.emit(
-                {
-                    tab: this.groups.toArray()[prevSelectedIndex].relatedTab,
-                    group: this.groups.toArray()[prevSelectedIndex]
-                });
-        }
-
-        this.selectedIndex = args.group.index;
-        this.groups.forEach((p) => {
-            if (p.index !== this.selectedIndex) {
-                this.deselectGroup(p);
+        if (this.hasContentTabs) {
+            this.contentTabs.forEach((theTab) => {
```

Comment:
Make sure we are not losing the tab selection or the selectedIndex in this scenario.

Reply:
Addressed.